toc/rfc/rfc-draft-add-cnbs.md

```diff
@@ -0,0 +1,194 @@
+# Meta
+[meta]: #meta
+- Name: Add Cloud Native Buildpacks
+- Start Date: 2023-05-01
+- Author: @gerg
+- Contributing Authors: @ryanmoran, @robdimsdale, @dsabeti, @aramprice, @dsboulder
+- Status: Draft
+- RFC Pull Request: #591
+
+## Summary
+
+[Cloud Native Buildpacks (CNBs)](https://buildpacks.io/), also known as v3
+buildpacks, are the current generation of buildpacks, and offer some
+improvements over the v2 buildpacks that CF Deployment currently uses. The
+Cloud Foundry Foundation already has an implementation of Cloud Native
+Buildpacks via the [Korifi](https://www.cloudfoundry.org/technology/korifi/)
+project, however these CNBs are not currently integrated with core CF
+Deployment.
+
+This RFC proposes adding CNB support to Cloud Foundry and including the Korifi
+buildpacks in CF Deployment.
+
+## Problem
+
+The v2 buildpacks are effectively in maintenance mode and do not receive
+substantial new features. By not integrating with v3 buildpacks, Cloud Foundry
+is missing out on new capabilities like application
+[SBOM](https://en.wikipedia.org/wiki/Software_supply_chain) generation, new
```

Comment:
Buildpacks produce SBOMs at known locations on the filesystem of the OCI image. It would probably be the responsibility of the shim layer to ensure these SBOM files are processed. I assume they wouldn't be packaged into the droplet, as that's just the executable, so they would have to provided to some Cloud Foundry handler via CF-specific metadata.

Maybe you knew all that already and your question was just around how CF will define, consume, and render this information. In which case I don't have any extra information :)

Reply:
Thanks, @robdimsdale ! Actually, I've been unclear on the specifics of how the SBOM data gets propagated even in a K8s context, especially so it can integrate with system such as Grype that can process the SBOM data. Does the OCI image metadata contain the SBOM, or do those processing systems have to unpack the filesystem layers in the image to get it from a canonical location? If the former, does kpack then propagate the SBOM data from the filesystem into the image metadata?

Anyway, I definitely think we can work out all the details of exposing CNB-generated SBOM data to the authorized set of CF users in due time, but it'd be great to refer to it at a high level as intended work in this part of the proposal.

Reply:
@robdimsdale Can you expand more on why you think the SBOM shouldn't be in the droplet? There are some potential advantages to bundling the SBOM with the executable. For example:
- Benefiting people who are scanning droplets directly from the blobstore (though maybe this won't be as common if we have built-in SBOM support).
- Preserving the SBOM when downloading/uploading droplets.

10k-100k lines per SBOM is probably too large to put in the CCDB, so we would likely have to store the SBOMs in the blobstore anyway.

Reply:
Maybe my mental model is wrong, but I was thinking the droplet was intended only to be the executable and its supporting filesystem. If there's already precedent for storing metadata in the droplet (or we're willing to create that precedent) then that would be fine. Storing the SBOM data in the droplet is closer to how the V3 lifecycle handles the SBOM data.

Reply:
And yes, I don't think you'd want to parse the SBOM data to store it in a structured format in CCDB. I was assuming there was already some way of storing large amounts of (potentially compressed) metadata already in CF/CAPI.

Reply:
TOC discussion summary: 
* out of scope to have them available for lifecycle
* can iterate on SBOMs stuff after first pass
* don't know what tooling we want 
* resolve to note in the RFC that the SBOM data will reside in the droplet

Reply:
https://github.com/cloudfoundry/community/commit/5a7ebf4f81696d4f4dcd6c40b9dcca8dc4597197